service.autosubs/default.py

```diff
@@ -0,0 +1,161 @@
+
+    ExcludePath4 = getSetting('ExcludePath4')
+    if ExcludePath4 and getSettingAsBool('ExcludePathOption4'):
+        if (movieFullPath.find(ExcludePath4) > -1):
+            Debug("isExcluded(): Video is playing from '%s', which is currently set as excluded path 4." % ExcludePath4)
+            return False
+
+    ExcludePath5 = getSetting('ExcludePath5')
+    if ExcludePath5 and getSettingAsBool('ExcludePathOption5'):
+        if (movieFullPath.find(ExcludePath5) > -1):
+            Debug("isExcluded(): Video is playing from '%s', which is currently set as excluded path 5." % ExcludePath5)
+            return False
+
+    return True
+
+
+class AutoSubsPlayer(xbmc.Player):
+    def __init__(self, *args, **kwargs):
+        xbmc.Player.__init__(self)
+        Debug("Initalized")
+        self.run = True
+
+    def onPlayBackStopped(self):
+        Debug("Stopped")
+        self.run = True
+
+    def onPlayBackEnded(self):
+        Debug("Ended")
+        self.run = True
+
+    def onPlayBackStarted(self):
+        if (xbmc.Player().isPlayingVideo()):
+            check_for_specific = (__addon__.getSetting('check_for_specific').lower() == 'true')
+            specific_language = (__addon__.getSetting('selected_language'))
+            specific_language = xbmc.convertLanguage(specific_language, xbmc.ISO_639_2)
+            try:
+                xbmc.sleep(3000)
+                if self.getSubtitles():
+                    Debug("Subtitles already present '%s'" % self.getSubtitles())
+                    self.run = False
+            except:
+                pass
+            if self.run:
+                movieFullPath = xbmc.Player().getPlayingFile()
+                Debug("movieFullPath '%s'" % movieFullPath)
+                xbmc.sleep(1000)
+                availableLangs = xbmc.Player().getAvailableSubtitleStreams()
+                Debug("availableLangs '%s'" % availableLangs)
+                totalTime = xbmc.Player().getTotalTime()
+                Debug("totalTime '%s'" % totalTime)
+                videoclipAlbum = ''
+                if getSettingAsBool('ExcludeVideoClip'):
+                    videoclipAlbum = xbmc.InfoTagMusic.getAlbum()
+                    Debug("videoclipAlbum '%s'" % videoclipAlbum)
+
+                if (totalTime > ExcludeTime and (not videoclipAlbum) and (
+                    (not xbmc.getCondVisibility("VideoPlayer.HasSubtitles")) or (
+                    check_for_specific and not specific_language in availableLangs)) and all(
+                            movieFullPath.find(v) <= -1 for v in ignore_words) and (isExcluded(movieFullPath))):
+                    self.run = False
+                    xbmc.sleep(1000)
+                    Debug('Started: AutoSearching for Subs')
+                    xbmc.executebuiltin('ActivateWindow(SubtitleSearch)')
+                else:
+                    Debug('Started: Subs found or Excluded')
+                    xbmc.Player().showSubtitles(True)
```

Comment:
please avoid creating Player() instances all the time. Re-use the class: `self.showSubtitles`

Reply:
ok